File: AUGUST-2025/04-08-2025/p904.py
class Solution:
    def totalFruit(self, fruits: List[int]) -> int:
        myset = []
        count = 0
        lastCons = 1
        maxi = 0
        for i in range(len(fruits)):


            if(fruits[i] not in myset):
                if len(myset)<2:
                    myset.append(fruits[i])
                    count+=1
                else:
                    count=lastCons+1
                    if myset[0] != fruits[i-1]:
                        myset[0] = fruits[i]
                    else:
                        myset[1] = fruits[i]
            else:
                count+=1
                
            if(i>0 and fruits[i]==fruits[i-1]): lastCons+=1
            else: lastCons = 1

            maxi = max(maxi, count)

        return maxi

File: AUGUST-2025/04-08-2025/p904.php
class Solution {

    /**
     * @param Integer[] $fruits
     * @return Integer
     */
    function totalFruit($fruits) {
        $n = count($fruits);
        if ($n < 3) {
            return $n;
        }

        $basket = [];
        $l = 0; 
        $r = 0;
        $count = 0;
        $res = 0;

        while ($r < $n) {
            $basket[$fruits[$r]] = ($basket[$fruits[$r]] ?? 0) + 1;
            //set left pointer only for the first fruit of second type
            if (count($basket) === 2 && $basket[$fruits[$r]] === 1) {
                $l = $r;
            } elseif (count($basket) > 2) {
                unset($basket);
                $res = max($res, $count);
                $count = 0;
                $r = $l;
                continue;
            }
            $count++;
            $r++;
        }

        return max($res, $count);
    }
}